### PR TITLE
Prefix ambiguity

### PR DESCRIPTION
**Pull request by gcurtis79**
Bot would detect any leading character as a prefix.

Added 'If' check to see if first character matched prefix in settings.

Both bots respond to the Kossel prefix:

prefix error

Kossel ignores FrankenCube prefix (FrankenCube unpatched for testing)

prefix fixed

### CODE REVIEW

**Comment by gcurtis79** [comment]
Hopefully I'm not making you crazy with the mini-patches. I love this plugin.

Also, awesome work on the new user/permissions design, hoped for something like that soon.

**Comment by cameroncros** [comment]
Sorry, what was the problem here? You closed the PR?

Also, if anything, i am grateful that you are fixing things for me, so don't hold back if you want to change something.

File: octoprint_discordremote/command.py [hunk 1]
```diff
@@ -42,25 +42,27 @@ def __init__(self, plugin):
...
-
-        if user and not self.check_perms(command_string, user):
-            return None, error_embed(author=self.plugin.get_printer_name(),
-                                     title="Permission Denied")
-
-        if command.get('params'):
-            return command['cmd'](parts)
-        else:
-            return command['cmd']()
+        if prefix_char == parts[0][:1]:
```
Comment (cameroncros) [inline octoprint_discordremote/command.py]: I'd prefer if you changed this to:
```
if prefix_char == parts[0][:1]:
    return None, None
```

Comment (cameroncros) [reply octoprint_discordremote/command.py]: Also, the unit tests need to be fixed up.

**Comment by gcurtis79** [comment]
It is failing continuous integration, but I'm not sure why. So I killed this one and started another with a modification, but it still fails.

I'm still kinda new to GitHub and didn't know I could just change the file mid-PR.

**Comment by cameroncros** [comment]
You can keep committing to the branch you created. I have a working fix if you would like me to just do it?

**Comment by gcurtis79** [comment]
That's fine. I'm honestly a rookie here.

The prefix check would be better in discord.py, but it didn't have the module for reading the settings so far as I could tell.

**Pull request closed without merging**